swift/ql/lib/codeql/swift/controlflow/internal/ControlFlowGraphImpl.qll

```diff
@@ -207,29 +207,70 @@ module Stmts {
     override FailStmt ast;
   }
 
-  private class StmtConditionTree extends AstPostOrderTree {
```

Comment:
Why the switch from PostOrder to PreOrder here?

Reply:
Good question! When it's a PreOrder the final `Completion` will just be whatever `Completion` of the children (i.e., a `BooleanCompletion` if the `StmtCondition` is a boolean expression, and a `MatchingCompletion` if the `StmtCondition` is a declaration with a pattern match). 

In contrast, if we keep it as a `PostOrder` we have to include all the logic about whether the child generates a `BooleanCompletion` or `MatchingCompletion` in the  [`isValidForSpecific`](https://github.com/github/codeql/blob/main/swift/ql/lib/codeql/swift/controlflow/internal/Completion.qll#L45) predicate.